### PR TITLE
Sync main with bugfixes on release/0.6.2-nci

### PR DESCRIPTION
**Pull request by tgeoghegan**
NCI has been running a `facilitator` forked from tag 0.6.2 for some time now, and I have been making targeted hotfixes on `release/0.6.2-nci` to minimize risk to NCI's deployment. This PR brings `main` to a state where it can safely be deployed to NCI and eventually MITRE's AWS deployments. See the individual commit messages for details on what changed and the motivation.

### CODE REVIEW

File: facilitator/src/aws_credentials.rs
Comment (aaomidi) [inline facilitator/src/aws_credentials.rs:79]: Might be more rusty to use: https://doc.rust-lang.org/std/default/trait.Default.html

**Review by aaomidi — APPROVED** [review]

File: facilitator/src/aws_credentials.rs
Comment (tgeoghegan) [reply facilitator/src/aws_credentials.rs:79]: The semantics are messy here: the default for this enum should probably be `new_web_identity_from_kubernetes_environment` to preserve backward compatibility, but `default` here refers to Rusoto's `DefaultCredentialsProvider`. This is one of those problems that I would classify as inherently stupid (we are forced by backward compat into making an API design that is bad), and thus any solution to it will reflect that stupidity.

Comment (aaomidi) [reply facilitator/src/aws_credentials.rs:79]: Love it! Sounds good to me. :shipit:

Comment (tgeoghegan) [reply facilitator/src/aws_credentials.rs:79]: The other thing I would note is that implementing `Default` is meant to be a convenience for the user of this enum, to spare them the hassle of filling in fields they don't care about or selecting an enum variant when they have no way of knowing which one they want. In this case, there is no such caller: we provide the `Provider::new()` associated function which does the right thing based on its inputs, so implementing `Default` wouldn't save anyone any hassle.

**Comment by tgeoghegan** [comment]
This change needs some more testing before I can merge it to main, to make sure that it behaves correctly in the presence of `TASK_QUEUE_USE_DEFAULT_AWS_CREDENTIALS_PROVIDER` (and the storage equivalents), so I will merge once I have enough confidence that it is safe.

**Pull request merged**
